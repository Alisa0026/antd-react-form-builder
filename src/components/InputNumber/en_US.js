export default {
  FormBuilderIntegerInput: {
    "formatErrorMsg": "please input integer",
  },
  FormBuilderFloatInput: {
    "formatErrorMsg": "please input decimals",
  },
}



