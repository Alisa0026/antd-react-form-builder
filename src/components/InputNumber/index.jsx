import React from 'react'
import {
  Input 
} from "antd"
import BasicItem from '../BasicItem'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'
import localeText from './zh_CN'

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class InputNumber extends React.Component {
    constructor(props){
      super(props);
    }

    static contextTypes = {
      antLocale: React.PropTypes.object,
    };
    //其他地方也可以访问这个rules，测试就要用到
    static getRules(replaceLocale) {
      var locale;
      if(replaceLocale){
        locale = replaceLocale;
      }else {
        locale = localeText;
      }
      if(!locale.FormBuilderFloatInput){
        locale.FormBuilderFloatInput = localeText.FormBuilderFloatInput;
      }
      if(!locale.FormBuilderIntegerInput){
        locale.FormBuilderIntegerInput = localeText.FormBuilderIntegerInput;
      }
      //这里运行了多次，没想到更好办法，目前先这样处理
      return {
        integer: [
          { 
            type: "integer",
            message: (
              locale.FormBuilderIntegerInput.formatErrorMsg ||
              localeText.FormBuilderIntegerInput.formatErrorMsg
            ),
          }
        ],
        float: [
          { 
            type: "float",
            message: (
              locale.FormBuilderFloatInput.formatErrorMsg ||
              localeText.FormBuilderFloatInput.formatErrorMsg
            ),
          }
        ], 
      }
    };

    //定义rule,type等信息 
    getInfoObject(type){
      let { locale } = this.props;
      if(!locale){
        locale = this.context.antLocale;
      }
      var rules = InputNumber.getRules(locale); 
      switch(type){
        case "float":
          return {
            rules: rules.float,
            type: "number",
          }
        case "integer":
          return {
            rules: rules.integer,
            type: "number",
          }
        default:
          return {
            type: "number",
            rules: [],
          };
      }
    }


    render(){
      let { type,rules,...other } = this.props;
      other.type = "number";
      if(other.value){
        other.value = Number(other.value);
      }
      other.targetComponent = Input;
      let infoObject = this.getInfoObject(type); 
      let temp_rules = [];
      Array.prototype.push.apply(temp_rules,rules)
      Array.prototype.push.apply(temp_rules,infoObject.rules || [])
      other.rules = temp_rules;
      other.type = infoObject.type;
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return InputNumber;
}

export default component(BasicItem) 





