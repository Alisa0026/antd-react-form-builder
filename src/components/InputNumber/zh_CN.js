export default {
  FormBuilderIntegerInput: {
    "formatErrorMsg": "请输入整数",
  },
  FormBuilderFloatInput: {
    "formatErrorMsg": "请输入小数",
  },
}


