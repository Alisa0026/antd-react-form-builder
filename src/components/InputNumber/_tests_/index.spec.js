import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import LocaleProvider from "antd/lib/locale-provider"
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import InputNumber from '../index'

describe('InputNumber',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <InputNumber 
        required
        className="test"
        label="name"
        name="test"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(InputNumber); 
  it("type='undefined' and value='ddd'，result value should be NaN",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        value="ddd"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().value).toEqual(NaN);
  })
  it("type='number' and value='ddd'，result value should be NaN",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        value="ddd"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    expect(antdInputMount.props().value).toEqual(NaN);
  })
  it("type='integer' should show error state",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        type="integer"
        value="1.1"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //有label时，className="ant-form-item-label"
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("type='integer' should show success state",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        type="integer"
        value="1"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //有label时，className="ant-form-item-label"
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
  it("type='float' should show error state",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        type="float"
        value="1"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //有label时，className="ant-form-item-label"
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-error");
  })
  it("type='float' should show success state",function(){
    const wrapper = mount(
      <InputNumber 
        name="test"
        type="float"
        value="1.1"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //有label时，className="ant-form-item-label"
    expect(antdFormItemMount.childAt(0).childAt(0).props().className).toEqual("ant-form-item-control has-success");
  })
})

