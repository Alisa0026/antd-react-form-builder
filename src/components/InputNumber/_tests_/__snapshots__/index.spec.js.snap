// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber renders correctly 1`] = `
<span>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-form-item-label"
    >
      <label
        class="ant-form-item-required"
        title="name"
      >
        name
      </label>
    </div>
    <div
      class="ant-form-item-control-wrapper"
    >
      <div
        class="ant-form-item-control "
      >
        <input
          class="ant-input test"
          name="test"
          type="number"
          value=""
        />
      </div>
    </div>
  </div>
</span>
`;
